apps/sms/test/marionette/lib/messages.js

```diff
@@ -208,27 +220,21 @@ var ConversationAccessor = require('./views/conversation/accessors');
         },
 
         performHeaderAction: function() {
-          this.Composer.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
+          this.Composer.headerActionButton.tap();
+          this.client.switchToShadowRoot();
         },
 
         performReportHeaderAction: function() {
-          this.Report.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
+          this.Report.headerActionButton.tap();
+          this.client.switchToShadowRoot();
         },
 
         performGroupHeaderAction: function() {
-          this.Participants.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
```

Comment:
__issue:__ same here, but ```this.Participants.header``` respectively. - it basically means you want to enter into Shadow Dom of Participant View header and tap on action button there. I'd recommend you to read more about Shadow DOM [1] - it's the future :)

[1] http://mdn.io/shadowdom